packages/fuels-macros/src/abigen/code_gen/abi_types.rs

```diff
@@ -118,7 +115,10 @@ impl FullABIFunction {
             abi_function.name.clone(),
             inputs,
             FullTypeApplication::from_counterpart(&abi_function.output, types),
-            abi_function.attributes.clone(),
+            abi_function
+                .attributes
+                .as_ref()
+                .map_or(vec![], Clone::clone),
```

Comment:
It's getting crowded, I'd extract the attributes into a separate variable:
```rust
        let attributes = abi_function.attributes.iter().flatten().cloned().collect();
        FullABIFunction::new(
            abi_function.name.clone(),
            inputs,
            FullTypeApplication::from_counterpart(&abi_function.output, types),
            attributes,
        )
```

Also the flatten->cloned->collect is another approach. I'm fine with both.